python/ml4ir/base/model/relevance_model.py

```diff
@@ -808,5 +809,47 @@ def calibrate(self, relevance_dataset, logger, logs_dir_local, temperature_init)
                                  dataset=relevance_dataset,
                                  logger=logger,
                                  logs_dir_local=logs_dir_local,
-                                 temperature_init=temperature_init,
-                                 file_io=self.file_io)
+                                 file_io=self.file_io,
+                                 **kwargs)
+
+    def add_temperature_layer(self, temperature: float = 1.0, layer_name: str = 'temperature_layer'):
+        """Add temperature layer to the input of last activation (softmax) layer
+        Parameters
+        ----------
+        self: RelevanceModel
+                        input RelevanceModel object that its last layer inputs will be divided by a
+                        temperature value
+        temperature: float
+                     a scalar value to scale the last activation layer inputs
+        layer_name: str
+            name of the temperature scaling layer
+        Returns
+        -------
+        `RelevanceModel`
+        updated RelevanceModel object with temperature
+        """
+
+        # get  last layer's output  --> MUST **NOT** BE AN ACTIVATION (e.g. SOFTMAX) LAYER
```

Comment:
But isn't typically our last layer an activation layer such as a softmax?
What would happen if that's the case?

Reply:
in our implementation we add the final activation (softmax) as a separate layer in the last layer. 
NN: Input, Dense1, Dense2, Activation ----- adding temperature scaling layer-----> NN: Dense1, Dense2, TS layer, Activation
I create a totally new functional API Model from the existing Model to add a new layer in a right place. I do not want to loose any layer and in the above example I do want Dense2 (without softmax) be right before TS layer because I want to scale the output of Dense2.

Reply:
Ok that makes sense, just to double check, it is currently working under the assumption we only have a softmax as our last activation function, right? Would we have to revisit it if we consider something else then https://github.com/salesforce/ml4ir/blob/master/python/ml4ir/applications/classification/config/keys.py#L7 (CATEGORICAL_CROSS_ENTROPY) ?

Reply:
sigmoid/softmax/or any Activation for the last output should be fine. The thing is that the activation should produce a probability vector (sum to 1) otherwise adding calibration does not make sense in theory because TS  is supposed to calibrate the classification probabilities.

Reply:
We should probably add that as comment/doc somewhere :))

Reply:
@Ullimague  yes, definitely. added here :
https://github.com/salesforce/ml4ir/pull/119/commits/afa20e84d414f72e33cc4330be6e9c5df8584932#diff-76516993b9207e862e3d127124d5549728d65b4ff22fc829dc4770226eae0cb4R29